ui/threads.js

```diff
@@ -87,6 +87,24 @@ module.exports = function (componentsState) {
         }
       },
 
+      pullRoots: function(categoryKey, messages) {
+        var self = this
+        var rootKeys = messages.map((x) => { return x.value.content.root || x.key })
+          .filter((x, index, self) => { return self.indexOf(x) == index })
```

Comment:
You can use a Set here instead of the filter (unique), it's faster and a bit easier to see what is going on.

Reply:
Haven't used Set before.  It's a little less readable to me, but take a look and see what you think.